File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length !== t.length) return false;
    const sLogger = {};
    const tLogger = {};

    for (let i = 0; i < s.length; i++) {
        if (!sLogger[s[i]]) sLogger[s[i]] = 1;
        else sLogger[s[i]]++;
        if (!tLogger[t[i]]) tLogger[t[i]] = 1;
        else tLogger[t[i]]++;
    };
    for (let key of Object.keys(sLogger)) {
        if (!tLogger[key]) return false;
        if (sLogger[key] !== tLogger[key]) return false;
    };
    return true;
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False

        sLogger = {}
        tLogger = {}

        for i in range(len(s)):
            sLogger[s[i]] = sLogger.get(s[i], 0) + 1
            tLogger[t[i]] = tLogger.get(t[i], 0) + 1
        
        return sLogger == tLogger
